File: idea-plugin/src/github/jk1/smtpidea/ui/EnumRadioGroup.java
package github.jk1.smtpidea.ui;

import javax.swing.*;
import java.util.HashMap;
import java.util.Map;

/**
 * Radio button group mapped to enum set. Simplifies
 * UI control bindings for enum-typed configuration properties.
 *
 * @author Evgeny Naumenko
 */
public class EnumRadioGroup<T extends Enum> extends ButtonGroup {

    private Map<T, AbstractButton> mapping = new HashMap<T, AbstractButton>();

    public void add(AbstractButton b, T value) {
        super.add(b);
        mapping.put(value, b);

    }

    public T getSelected() {
        for (Map.Entry<T, AbstractButton> entry : mapping.entrySet()) {
            if (entry.getValue().isSelected()) {
                return entry.getKey();
            }
        }
        throw new IllegalStateException("No selected value found for EnumRadioGroup");
    }

    public void setSelected(T value) {
        mapping.get(value).setSelected(true);
    }
}

File: src/main/kotlin/github/jk1/smtpidea/ui/EnumRadioGroup.kt
package github.jk1.smtpidea.ui

import javax.swing.ButtonGroup
import javax.swing.AbstractButton
import java.util.HashMap

/**
 * Radio button group mapped to enum set. Simplifies
 * UI control bindings for enum-typed configuration properties.
 */
public class EnumRadioGroup<T> : ButtonGroup(){

    private var mapping: MutableMap<T, AbstractButton> = HashMap<T, AbstractButton>()

    public fun add(b: AbstractButton, value: T) {
        super.add(b)
        mapping.put(value, b)
    }

    public fun getSelected(): T {
        for (entry in mapping.entrySet()) {
            if (entry.getValue().isSelected()) {
                return entry.getKey()
            }
        }
        throw IllegalStateException("No selected value found for EnumRadioGroup")
    }

    public fun setSelected(value: T): Unit = mapping.get(value)?.setSelected(true)
}